pkg/operation/botanist/dns.go

```diff
@@ -220,9 +219,13 @@ func (b *Botanist) waitUntilDNSProviderReady(ctx context.Context, name string) e
 		if msg := provider.Status.Message; msg != nil {
 			message = *msg
 		}
+		providerErr := fmt.Errorf("DNS provider %q is not ready (status=%s, message=%s)", name, status, message)
 
 		b.Logger.Infof("Waiting for %q DNS provider to be ready... (status=%s, message=%s)", name, status, message)
-		return retry.MinorError(fmt.Errorf("DNS provider %q is not ready (status=%s, message=%s)", name, status, message))
+		if status == dnsv1alpha1.STATE_ERROR || status == dnsv1alpha1.STATE_INVALID {
+			return retry.SevereError(providerErr)
+		}
+		return retry.MinorError(providerErr)
```

Comment:
We missed adapting the `WaitForDNSEntryReady` function accordingly, can you change as well similar to this change?
https://github.com/gardener/gardener/pull/2336/files#diff-33d2a8cacfcd2dda84699a11862f3ab0L274-L282

Reply:
Nice catch!